lib/rules.js

```diff
@@ -197,9 +197,11 @@ module.exports = [
           style = styleForToken;
         }
         if ((styleForToken !== style) &&
-            !((style === "setext_with_atx") &&
+            !((style === "setext_with_atx" || style === "setext_with_atx_closed") &&
               (/h[12]/.test(token.tag) && (styleForToken === "setext")) ||
-               (/h[^12]/.test(token.tag) && (styleForToken === "atx")))) {
+              (/h[^12]/.test(token.tag) &&
+                  (styleForToken === "atx" || styleForToken === "atx_closed")))
```

Comment:
Line 201 and 202, and actually 200 as well, are all siblings, so I lined them all up on their opening paren. Is it still expected that line 202 should be indented one space to the right?

Regarding 203, sorry, my default tab is four spaces. Line 203 is a nested condition. Would you like it 1 or or 2 spaces in from the previous line?